numpy/lib/function_base.py

```diff
@@ -990,15 +997,37 @@ def gradient(f, *varargs, **kwargs):
     >>> y = x**2
     >>> np.gradient(y, dx, edge_order=2)
     array([-0.,  2.,  4.,  6.,  8.])
+
+    The axis keyword can be used to specify a subset of axes of which the gradient is calculated
+    >>> np.gradient(np.array([[1, 2, 6], [3, 4, 5]], dtype=np.float), axis=0)
+    array([[ 2.,  2., -1.],
+           [ 2.,  2., -1.]])
     """
     f = np.asanyarray(f)
     N = len(f.shape)  # number of dimensions
+
+
+    axes = kwargs.pop('axis', tuple(range(N)))
+    # check axes to have correct type and no duplicate entries
+    if type(axes) is int:
+        axes = (axes,)
+    if type(axes) is not tuple:
+        raise TypeError("A tuple of integers or a single integer is required")
+
+    #normalize axis values:
+    axes = tuple([ x + N if x < 0 else x for x in axes])
```

Comment:
PEP8 -- you have one extra space here after `tuple([`

You also don't need to create the intermediate list. Just omit the brackets and create the tuple directly using a generator expression: `tuple(x + N if x < 0 else x for x in axes)`